docs/zos-native-os-linkage.md

```diff
@@ -0,0 +1,61 @@
+# z/OS Native OS Linkage Example
+
+These steps describe how to build and test this sample REST API in a way that
+uses JNI to call a native z/OS service - WTO.  You can follow these steps and modify them to
+call your own assembler / C API.
+
+These instructions assume that you follow the [z/OS deployment setup](./zos-deployment).
+
+## JNI Overview
+
+There are many tutorials online to describe details and examples of JNI.  
+
+At a high level, the process involves making use of the `native` keyword in a class.  After this, you run the `javah` command against the class to emit a C/C++ header file.  This emitted header file will contain function prototypes for the native method for which you must provide the implementation.
+
+Here is the example `javah` command for this project:
```

Comment:
Just an FYI, At least with oracle, etc. `javah` is no longer available I believe after JDK 8. I think it's replaced by a flag on the compiler.

Reply:
Good point - I didn't know this.  I made a note in the markdown - maybe in a future PR the `makefile` and sample command will be updated to use the flag.